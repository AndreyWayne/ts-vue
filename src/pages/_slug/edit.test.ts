import { mount } from '@vue/test-utils'
import Edit from './edit.vue'

function build ({ title = 'Foo' } = {}, component: any = Edit) {
  const wrapper = mount(component, {
    propsData: {
      material: {
        title
      }
    }
  })

  const vm = wrapper.vm

  return {
    vm,
    wrapper
  }
}

describe('Edit page', () => {
  it('should render', () => {
    const { wrapper } = build()

    expect(wrapper).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('get currentAttributes', () => {
    it('should return material attributes', () => {
      const { vm, wrapper } = build()

      const material = {
        attributes: {
          title: 'Foo'
        }
      }

      wrapper.setProps({ material })
      expect((vm as any).currentAttributes).toEqual(material.attributes)
    })

    it('should return an empty object', () => {
      const { vm, wrapper } = build()

      const material = {}

      wrapper.setProps({ material })
      expect((vm as any).currentAttributes).toEqual(material)
    })
  })

  describe('autoResize', () => {
    it('should set element style', () => {
      const { vm }: { vm: any } = build()

      const element = document.createElement('textarea')
      vm.autoResize(element)

      expect(element.style.height).toMatchSnapshot()
    })

    it('should ignore', () => {
      const { vm }: { vm: any } = build()
      vm.autoResize()
    })
  })

  describe('onTextAreaInput', () => {
    it('should call autoResize', () => {
      const { vm }: { vm: any } = build()

      vm.autoResize = jest.fn()

      const event = new Event('input')
      vm.onTextAreaInput(event)

      expect(vm.autoResize).toBeCalled()
    })
  })
})
