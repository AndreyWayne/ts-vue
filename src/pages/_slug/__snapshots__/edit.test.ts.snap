// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit page autoResize should set element style 1`] = `"0px"`;

exports[`Edit page should render 1`] = `
"<div class=\\"\\">
  <div class=\\"\\"><textarea placeholder=\\"Заголовок\\" rows=\\"1\\" class=\\"\\"></textarea>
    <!---->
    <!---->
  </div>
</div>"
`;
